glusterd2/transaction/lock.go

```diff
@@ -149,15 +149,15 @@ func CreateLockFuncs(key string) (LockUnlockFunc, LockUnlockFunc) {
 	return lockFunc, unlockFunc
 }
 
-func (t *Txn) lock(lockID string) error {
+// Locks are the collection of cluster wide transaction lock
+type Locks map[string]*concurrency.Mutex
+
+func (l Locks) lock(lockID string) error {
 	// Ensure that no prior lock exists for the given lockID in this transaction
-	if _, ok := t.locks[lockID]; ok {
+	if _, ok := l[lockID]; ok {
 		return ErrLockExists
 	}
 
-	logger := t.Ctx.Logger().WithField("lockID", lockID)
```

Comment:
Restore the logs here. They can be helpful.

Reply:
restored the previous logs